app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -104,7 +131,19 @@ def with_provider_connection(options = {})
       connection = connect(options)
       yield connection
     ensure
-      connection.try(:disconnect) rescue nil
+      # The `connect` method will return different types of objects depending on the value of the `version`
+      # parameter. If `version` is 3 the connection object is created by the `ovirt` gem and it is closed
+      # using the `disconnect` method. If `version` is 4 the object is created by the oVirt Ruby SDK, and
+      # it is clused using the `close` method.
+      begin
+        if connection.respond_to?(:disconnect)
+          connection.disconnect
+        elsif connection.respond_to?(:close)
+          connection.close
+        end
+      rescue
```

Comment:
Not sure if I can be more specific, the code used to be `connection.try(:disconnect) rescue nil` and that also discards every error, if I understand correctly, and I don't want to change the behaviour. However, I can certainly log the error, I will do it.